core/domain/rights_manager_test.py

```diff
@@ -450,6 +452,112 @@ def test_get_multiple_exploration_rights(self):
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b, rights_manager.ROLE_OWNER)
+        exp_rights = rights_manager.get_exploration_rights(self.EXP_ID)
+
+        self.assertTrue(exp_rights.is_owner(self.user_id_b))
+
+    def test_assign_voice_artist_role_owner(self):
+        exp = exp_domain.Exploration.create_default_exploration(self.EXP_ID)
+        exp_services.save_new_exploration(self.user_id_a, exp)
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b,
+            rights_manager.ROLE_VOICE_ARTIST)
+        exp_rights = rights_manager.get_exploration_rights(self.EXP_ID)
+
+        self.assertFalse(exp_rights.is_owner(self.user_id_b))
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b, rights_manager.ROLE_OWNER)
+        exp_rights = rights_manager.get_exploration_rights(self.EXP_ID)
+
+        self.assertTrue(exp_rights.is_owner(self.user_id_b))
+
+    def test_editor_cannot_be_assigned_role_editor(self):
+        exp = exp_domain.Exploration.create_default_exploration(self.EXP_ID)
+        exp_services.save_new_exploration(self.user_id_a, exp)
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b,
+            rights_manager.ROLE_EDITOR)
+
+        with self.assertRaisesRegexp(
+            Exception, 'This user already can edit this'):
+            rights_manager.assign_role_for_exploration(
+                self.user_a, self.EXP_ID, self.user_id_b,
+                rights_manager.ROLE_EDITOR)
+
+    def test_voice_artist_cannot_be_assigned_role_voice_artist(self):
+        exp = exp_domain.Exploration.create_default_exploration(self.EXP_ID)
+        exp_services.save_new_exploration(self.user_id_a, exp)
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b,
+            rights_manager.ROLE_VOICE_ARTIST)
+
+        with self.assertRaisesRegexp(
+            Exception, 'This user already can voiceover this'):
+            rights_manager.assign_role_for_exploration(
+                self.user_a, self.EXP_ID, self.user_id_b,
+                rights_manager.ROLE_VOICE_ARTIST)
+
+    def test_viewer_cannot_be_assigned_role_viewer(self):
+        exp = exp_domain.Exploration.create_default_exploration(self.EXP_ID)
+        exp_services.save_new_exploration(self.user_id_a, exp)
+
+        rights_manager.assign_role_for_exploration(
+            self.user_a, self.EXP_ID, self.user_id_b,
+            rights_manager.ROLE_VIEWER)
+
+        with self.assertRaisesRegexp(
+            Exception, 'This user already can view this'):
+            rights_manager.assign_role_for_exploration(
+                self.user_a, self.EXP_ID, self.user_id_b,
+                rights_manager.ROLE_VIEWER)
+
+    def test_public_explorations_cannot_be_assigned_role_viewer(self):
```

Comment:
Maybe `test_public_explorations_cannot_reassign_viewers`?

Reply:
We are not reassigning but we are assigning for the first time